src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -47,10 +61,131 @@ public partial class MediaManager : Java.Lang.Object, IPlayer.IListener
 			ControllerAutoShow = false,
 			LayoutParameters = new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent, GravityFlags.CenterHorizontal)
 		};
-
+		Task Result = CheckAndRequestForeGroundPermission(CancellationToken.None);
+		if(Result.IsFaulted)
```

Comment:
No this isn't good... At the time that you check if the Task is faulted, it couldn't be started yet... What I suggest is to have a local field `Task checkPermissionTask` and call `await checkPermissionTask` inside the method that will do something that needs that permission. Does it make sense to you?

(Yeah, I know... sync over async is hard)